src/map/battlefield.cpp

```diff
@@ -463,9 +463,10 @@ bool CBattlefield::RemoveEntity(CBaseEntity* PEntity, uint8 leavecode)
                 if (static_cast<CPetEntity*>(PEntity)->isAlive() && PEntity->PAI->IsSpawned())
                     static_cast<CPetEntity*>(PEntity)->Die();
 
-                m_AllyList.erase(std::remove_if(m_AllyList.begin(), m_AllyList.end(), check), m_AllyList.end());
-                GetZone()->DeletePET(PEntity);
-                delete PEntity;
```

Comment:
Are you sure it is not needed to delete the PEntity when removed from the battlefield?

Reply:
correct because the zone_entities.cpp handles the deletion already.
but I just saw that this function can be called from Lua.
so i will add a line "PEntity->status = STATUS_DISAPPEAR".
This will allow the zone_entities.cpp to pick up the deletion correctly and remove the Pet.